main/php_open_temporary_file.c

```diff
@@ -303,7 +303,16 @@ PHPAPI int php_open_temporary_fd_ex(const char *dir, const char *pfx, zend_strin
 		if (temp_dir &&
 		    *temp_dir != '\0' &&
 		    (!(flags & PHP_TMP_FILE_OPEN_BASEDIR_CHECK_ON_FALLBACK) || !php_check_open_basedir(temp_dir))) {
-			return php_do_open_temporary_file(temp_dir, pfx, opened_path_p);
+			fd = php_do_open_temporary_file(temp_dir, pfx, opened_path_p);
+			if (!(flags & PHP_TMP_FILE_SILENT) && dir && *dir != '\0') {
+				/* Default temporary directory fallback. */
+				if (UNEXPECTED(fd == -1)) {
+					/* TODO: decide whether we should notice that even the fallback failed? */
+				} else {
```

Comment:
I just checked and all callers trigger their own warning if the function fails so I think you should remove this TODO. Also if we really wanted such warning / notice, it should happen for non fallback as well...